File: 프로그래머스/코딩테스트 입문 캘린더/Day 22/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    l = [0 for _ in range(201)]
    count = 0
    
    for line in lines:
        for i in range(line[0], line[1]):
            l[i+100] += 1
            
    for i in l:
        if i > 1:
            count += 1
            
    return count

File: 프로그래머스/lv0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.js
function solution(lines) {
  let DICT = {};
  lines.forEach(([start, end]) => {
    for (let i = start; i < end; i++) {
      DICT[i] = DICT[i] ? DICT[i] + 1 : 1;
    }
  });
  return Object.entries(DICT).filter(([_, v]) => v > 1).length;
}